src/test/java/com/puppycrawl/tools/checkstyle/internal/CliOptionsXdocsSyncTest.java

```diff
@@ -0,0 +1,111 @@
+import org.w3c.dom.Document;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+
+import com.puppycrawl.tools.checkstyle.internal.utils.XmlUtil;
+import picocli.CommandLine.Option;
+
+public class CliOptionsXdocsSyncTest {
+
+    @Test
+    public void validateCliDocSection() throws Exception {
+        final Map<String, String> cmdDesc = new HashMap<>();
+        final Set<String> ignoreXdocToMain = new HashSet<>();
+        // used to ignore options present in xdocs but not in Main.java
+        ignoreXdocToMain.add("-V");
+        ignoreXdocToMain.add("-h");
+
+        final NodeList sections = getSectionsFromXdoc("src/xdocs/cmdline.xml.vm");
+
+        for (int i = 0; i < sections.getLength(); i++) {
+            final Node subSection = sections.item(i);
+            final Node name = subSection.getAttributes().getNamedItem("name");
+            if ("Command line usage".equals(name.getNodeValue())) {
+                final Set<String> cmdOptions = getListById(subSection, "CLI_Options");
+                for (String option : cmdOptions) {
+                    final String text = option.trim().replaceAll("\\s+", " ");
+                    cmdDesc.put(text.substring(0, 2), text.substring(text.indexOf(" - ") + 3));
+                }
+            }
+        }
+
+        final Class<?> cliOptionsClazz = Class.forName("com.puppycrawl.tools.checkstyle"
+                + ".Main$CliOptions");
+        final Field[] fields = cliOptionsClazz.getDeclaredFields();
+        for (Field field : fields) {
+            final Option optionsAnnotation = field.getAnnotation(Option.class);
+            if (optionsAnnotation != null) {
+                final String option = optionsAnnotation.names()[0];
+                final String descXdoc = cmdDesc.get(option);
+                final String descMain = optionsAnnotation.description()[0];
+                assertWithMessage("CLI Option" + option + "present in "
+                        + "Main.java not documented in cmdline.xml.vm")
+                        .that(descXdoc).isNotNull();
+                assertWithMessage("CLI options descriptions in xdoc: " + descXdoc
+                        + " should match that of in Main.java: " + descMain)
+                        .that(descXdoc.equals(descMain)).isTrue();
+                cmdDesc.remove(option);
+            }
+        }
+
+        cmdDesc.entrySet().removeIf(entry -> ignoreXdocToMain.contains(entry.getKey()));
+        assertWithMessage("CLI Options" + cmdDesc + "present in "
+                + "cmdline.xml.vm not documented in Main.java")
+                .that(cmdDesc.isEmpty());
+    }
+
+    private static NodeList getSectionsFromXdoc(String xdocPath) throws Exception {
+        final Path path = Paths.get(xdocPath);
+        final String input = new String(Files.readAllBytes(path), UTF_8);
+
+        final Document document = XmlUtil.getRawXml(path.getFileName().toString(), input, input);
+        return document.getElementsByTagName("section");
```

Comment:
@gaurabdg This is still not done.

Reply:
@rnveach `document.getElementById` is returning `null`, since we have to define a separate DTD schema to specify which is actually the ID for this xml. https://stackoverflow.com/a/53044340

Reply:
If we can't get the element by ID, then we don't need the ID in the XDOC.
> `<ul id="CLI_Options">`

Reply:
@rnveach It is required for `getListById` method.